packages/toolpad-app/src/toolpad/AppEditor/PageEditor/RenderPanel/RenderOverlay.tsx

```diff
@@ -875,15 +877,29 @@ export default function RenderOverlay({ canvasHostRef }: RenderOverlayProps) {
             ? appDom.getNewFirstParentIndexInNode(dom, dragOverNode, 'children')
             : appDom.getNewLastParentIndexInNode(dom, dragOverNode, 'children');
 
-        if (!isPageRow(draggedNode)) {
-          const rowContainer = appDom.createElement(dom, PAGE_ROW_COMPONENT_ID, {});
-          domApi.addNode(rowContainer, dragOverNode, 'children', newParentIndex);
-          parent = rowContainer;
+        domApi.update((draft) => {
```

Comment:
Looks awesome, so we could use this kind of logic to batch changes under the same update without applying each one separately? Might also be a good way to abstract some of this logic, make it more readable and split it in different logical parts, maybe by moving these functions to a separate file?

Reply:
I'll probably go from this once we work on undo/redo, thanks for suggesting it!

Reply:
Also meshes quite well with `immer` if we start using it

Reply:
Another important advantage is that it takes away an annoying limitation of `domApi`, it's that it doesn't update the local `dom` variable until after the next render:

```tsx
const dom = useDom()
const domApi = useDomApi()
const doSomething = (someNode) => {
  domApi.addNode(someNode)
  // dom is not up to date so this returns false:
  if (!!appDom.getMaybeNode(dom, node.id)) {

  }
}

// versus

const dom = useDom()
const domApi = useDomApi()
const doSomething = (someNode) => {
  domApi.update(draft => {
    draft = appDom.addNode(draft, someNode)
    // draft is up to date so this returns true:
    if (!!appDom.getMaybeNode(draft, node.id)) {
      
    }
  })
}
```

Reply:
Okay, might require some changes in the drag & drop logic as in some places I've been assuming that the DOM is not updated yet, but anyway this should make that logic much more intuitive.